sentry-android-core/src/main/java/io/sentry/android/core/AndroidLogger.java

```diff
@@ -11,18 +11,28 @@
 
   @SuppressWarnings("AnnotateFormatMethod")
   @Override
-  public void log(SentryLevel level, String message, Object... args) {
+  public void log(
+      final @Nullable SentryLevel level,
+      final @Nullable String message,
```

Comment:
agreed, the only problem is if we log the result of something instead of hardcoded value, eg, the response from the API, we'd need to check NPE before calling the method `log`, its a trade-off, I guess making it nullable makes it dev. friendlier, but doing NPE check within the method has its performance impacts too.